.chloggen/deprecateallconfig.yaml

```diff
@@ -0,0 +1,30 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: config
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate all types and funcs in `config` package
+
+# One or more tracking issues or pull requests related to the change
+issues: [6422]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  - config.Type => component.Type
+  - config.DataType => component.DataType
+  - config.Receiver => component.ReceiverConfig
+  - config.UnmarshalReceiver => component.UnmarshalReceiverConfig
+  - config.[New]ReceiverSettings => component.[New]ReceiverConfigSettings
```

Comment:
The "setting" pattern is an old pattern we used to do "inheritance". So not enough since these "settings" are the struct that MUST be embedded in every other struct that implements config.

I am not trying to fix that pattern yet (will discuss later), so for the moment I kept that way since these "settings" belong to the "config".

Reply:
I understand this is an existing pattern here, and i think this change is in the right direction.

I do think that *if* we plan on getting rid of the `ConfigSettings` suffix, making this change in the same version would be preferable, otherwise all the components will have to change from `[Receiver|Exporter|Processor|Extension]Settings` to `[Receiver|Exporter|Processor|Extension]ConfigSettings` and then to whatever else is decided on.

Reply:
I think to remove that, we need to remove more functionality from Config. See https://github.com/open-telemetry/opentelemetry-collector/pull/5870, otherwise we need too many duplicate code to implement that interface.

Reply:
I want to get rid of the `*ConfigSettings` structs not suffix. I am lost about your comment a bit.

Reply:
Are you suggesting to not move yet the "*Setting" structs, and better work on removing these? If that is your suggestion, I think it is reasonable, and I can give a try.

Reply:
Reverted that change for the moment.